libbeat/docs/https.asciidoc

```diff
@@ -10,66 +10,150 @@
+operating system where {beatname_uc} is running. If the certificate authority that signed your node certificates
+is not in the host system's trusted certificate authorities list, you need
+to add the path to the `.pem` file that contains your CA's certificate to the
+{beatname_uc} configuration. This will configure {beatname_uc} to use a specific list of
+CA certificates instead of the default list from the OS.
++
+Here is an example configuration:
++
+--
 ["source","yaml",subs="attributes,callouts"]
 ----------------------------------------------------------------------
 output.elasticsearch:
-  username: {beatname_lc}
-  password: verysecret
-  protocol: https
-  hosts: ["elasticsearch.example.com:9200"]
+  hosts: ["localhost:9200"]
   ssl.certificate_authorities: <1>
     - /etc/pki/my_root_ca.pem
     - /etc/pki/my_other_ca.pem
   ssl.certificate: "/etc/pki/client.pem" <2>
   ssl.key: "/etc/pki/key.pem" <3>
 ----------------------------------------------------------------------
-<1> The list of CA certificates to trust
+<1> Specify the path to the local `.pem` file that contains your Certificate
+Authority's certificate. This is needed if you use your own CA to sign your node certificates.
 <2> The path to the certificate for SSL client authentication
 <3> The client certificate key
-
+--
++
 NOTE: For any given connection, the SSL/TLS certificates must have a subject
 that matches the value specified for `hosts`, or the SSL handshake fails.
 For example, if you specify `hosts: ["foobar:9200"]`, the certificate MUST
 include `foobar` in the subject (`CN=foobar`) or as a subject alternative name
 (SAN). Make sure the hostname resolves to the correct IP address. If no DNS is available, then
 you can associate the IP address with your hostname in `/etc/hosts`
 (on Unix) or `C:\Windows\System32\drivers\etc\hosts` (on Windows).
+
+ifndef::apm-server[]
+[role="xpack"]
+[float]
+[[securing-communication-kibana]]
+=== Secure communication with the Kibana endpoint
+
+If you've configured the <<setup-kibana-endpoint,{kib} endpoint>>,
+you can also specify credentials for authenticating with {kib} under `kibana.setup`.
+If no credentials are specified, Kibana will use the configured authentication method
+in the Elasticsearch output.
+
+For example, specify a unique username and password to connect to Kibana like this:
+
+--
+["source","yaml",subs="attributes,callouts"]
+----
+setup.kibana:
+  host: "mykibanahost:5601"
+  username: "{beat_default_index_prefix}_kib_setup" <1>
+  password: "{pwd}" <2>
+----
+<1> This user needs privileges required to set up dashboards. To create a user like this,
+see <<privileges-to-setup-beats>>.
+<2> This example shows a hard-coded password, but you should store sensitive
+values
+ifndef::serverless[]
+in the <<keystore,secrets keystore>>.
+endif::[]
+ifdef::serverless[]
+in environment variables.
+endif::[]
+endif::apm-server[]
+--
```

Comment:
Should this content move to its own top-level heading/page? The TOC would then include:

* Secure communication with Elasticsearch
* Secure communication with Kibana
* Secure communication with Logstash

Alternatively, we could group these under a new header, like this:
* Secure communication with the Elastic Stack
  * Elasticsearch
  * Kibana
  * Logstash

Reply:
I do like the idea of grouping the content under a new heading, "Secure communication with the Elastic Stack".